napari/_tests/test_with_screenshot.py

```diff
@@ -59,6 +60,7 @@ def test_z_order_adding_removing_images(make_napari_viewer):
 def test_z_order_images(make_napari_viewer):
     """Test changing order of images changes z order in display."""
     data = np.ones((11, 11))
+    data[0, 0] = 0
```

Comment:
These changes are kind of unreadable, in the sense that it's hard to understand the intent. (And the intent has nothing to do with the test itself anyway, it's just a byproduct of the new automatic clim calculation.)

To me, it shows that something is broken with the new convention of minval-1/maxval+1. I think np.zeros should always be shown as black, and np.ones should always be shown as white. I think it's also fine for e.g. np.full((5, 5), 5) to be shown as white, and np.full((5, 5), -5) to be shown as black.

The confusing part in those situations is that the clims *range* don't encompass the data. Therefore, I suggest the following when minval==maxval:

clim = (0, 1)
clim_range = (min(0, minval), max(1, maxval))

This would mostly preserve current behaviour while fixing a bunch of edge cases. I think it fixes all the edge cases *without* causing the weird situation that imshow(np.zeros(shape)) gives the same display as imshow(np.ones(shape)).

Reply:
fixed